cmake/OpenCVDownload.cmake

```diff
@@ -37,6 +37,47 @@ file(WRITE "${OPENCV_DOWNLOAD_LOG}" "#use_cache \"${OPENCV_DOWNLOAD_PATH}\"\n")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_CURL}")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_WGET}")
 
+# host
+set(OPENCV_DOWNLOAD_HOST "GITHUB" CACHE INTERNAL "Used for 3rdparty archives")
+# mirrors
+set(OPENCV_MIRROR_GITCODE "gitcode.net" CACHE INTERNAL "Link to mirror hosted by gitcode")
+set(OPENCV_MIRROR_CUSTOM "" CACHE INTERNAL "Link to another mirror hosted in gitlab-style")
+
+function(ocv_init_download)
+  # Run `git remote get-url origin` to get remote source
+  execute_process(
+    COMMAND
+      git remote get-url origin
+    RESULT_VARIABLE
+      RESULT_STATUS
+    OUTPUT_VARIABLE
+      OCV_GIT_ORIGIN_URL_OUT
+    ERROR_QUIET
+  )
```

Comment:
This can't work properly without working directory.

Reply:
I added
```
WORKING_DIRECTORY
      ${CMAKE_SOURCE_DIR}
```
in `execute_process`. Did this solve the problem?